src/Tooltip2.tsx

```diff
@@ -0,0 +1,402 @@
+        // Has trigger element or any of its children interactive elements?
+        const hasInteractiveChild = Array.from(triggerChildren).some(child => isFocusable(child as HTMLElement))
+        invariant(
+          isTriggerInteractive || hasInteractiveChild,
+          'The `Tooltip2` component expects a single React element that contains interactive content. Consider using a `<button>` or equivalent interactive element instead.',
+        )
+      }
+    })
+  }
+
+  const triggerEvtHandlers = {
+    onFocus: () => setOpen(true),
+    onBlur: () => {
+      setOpen(false)
+    },
+    onMouseEnter: () => {
+      setOpen(true)
+    },
+  }
+
+  // Make sure to compose the default event handlers for tooltip trigger with the ones that are passed in
+  function composeEventHandlers(child: React.ReactElement) {
+    const {onBlur: _onBlur, onFocus: _onFocus, onMouseEnter: _onMouseEnter} = child.props
+
+    return {
+      onBlur: () => {
+        _onBlur && _onBlur()
+        triggerEvtHandlers.onBlur()
+      },
+      onFocus: () => {
+        _onFocus && _onFocus()
+        triggerEvtHandlers.onFocus()
+      },
+      onMouseEnter: () => {
+        _onMouseEnter && _onMouseEnter()
+        triggerEvtHandlers.onMouseEnter()
+      },
+    }
+  }
+
+  function onKeyDown(e: React.KeyboardEvent) {
+    if (open && e.key === 'Escape') {
+      e.stopPropagation()
+      setOpen(false)
+    }
+  }
+
+  return (
+    <Box
+      ref={containerRef}
+      sx={{position: 'relative', display: 'inline-block'}}
+      onKeyDown={onKeyDown}
+      onMouseLeave={() => setOpen(false)}
+    >
+      {React.cloneElement(child as React.ReactElement<TriggerPropsType>, {
+        ...{
+          // if it is a type description, we use tooltip to describe the trigger
+          'aria-describedby': type === 'description' ? id : undefined,
+          // If it is a type description, we should keep the aria label if it exists, otherwise we remove it because we will use aria-labelledby
+          'aria-label': type === 'description' ? (children as React.ReactElement).props['aria-label'] : undefined,
+          //   If it is a label type, we use tooltip to label the trigger
+          'aria-labelledby': type === 'label' ? id : undefined,
+        },
+        ...composeEventHandlers(child as React.ReactElement<TriggerPropsType>),
+      })}
+
+      <Tooltip
+        data-direction={direction}
+        data-state={open ? 'open' : undefined}
+        data-align={align}
+        data-wrap={wrap}
+        data-delay={noDelay}
+        {...rest}
+        // Only need tooltip role if the tooltip is a description for supplementary information
+        role={type === 'description' ? 'tooltip' : undefined}
+        // stop AT from announcing the tooltip twice when it is a label type because it will be announced with "aria-labelledby"
+        aria-hidden={type === 'label' ? true : undefined}
+        id={id}
+      >
+        {text || label}
```

Comment:
This is such a boring suggestion but great usage for `??` here 😂 

```suggestion
        {text ?? label}
```